tests/components/iotawatt/test_sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Test setting up sensors."""
+from datetime import timedelta
+
+from homeassistant.components.sensor import (
+    ATTR_STATE_CLASS,
+    DEVICE_CLASS_ENERGY,
+    STATE_CLASS_TOTAL_INCREASING,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    ATTR_UNIT_OF_MEASUREMENT,
+    ENERGY_WATT_HOUR,
+)
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from . import INPUT_SENSOR, OUTPUT_SENSOR
+
+from tests.common import async_fire_time_changed
+
+
+async def test_sensor_add_remove(hass, mock_iotawatt):
+    """Test sensor adding and removing works."""
+    assert await async_setup_component(hass, "iotawatt", {})
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_entity_ids()) == 0
+
+    mock_iotawatt.getSensors.return_value["sensors"]["my_sensor_key"] = INPUT_SENSOR
+    async_fire_time_changed(hass, utcnow() + timedelta(seconds=30))
+    await hass.async_block_till_done()
```

Comment:
If we expect this to add a sensor I'd assert the state here before moving on and removing sensors below.